nodestream/pipeline/pipeline.py

```diff
@@ -144,6 +144,10 @@ def pipeline_has_died(self) -> bool:
 
     async def submit_object_or_die_trying(self, obj):
         should_continue = True
+        if self.pipeline_has_died():
```

Comment:
I was expecting this to be in the loop before the await. What was the logic behind putting this in the head?